packages/nimble-components/src/table/specs/table-spec.md

```diff
@@ -0,0 +1,242 @@
+
+Blazor support should be accomplished through the typical integration patterns. It should be noted that if the Nimble `Table` component is generic, then the Blazor component will mirror that generic API.
+
+---
+
+## Implementation
+
+The `nimble-table` will require various sub-components to exist in order to properly/efficiently render the data. The web components and their respective responsibilities are as follows:
+
+`<nimble-table>`
+
+-   Creates the row elements that will be used to render each row of data
+-   Provides any necessary elements for proper virtualization
+-   Manages keyboard navigation between rows
+-   Supports slotting of column elements
+
+`<nimble-table-row>`
+
+-   Creates the cell elements that will be used to render each value in the data
+-   Manages keyboarding within the row
+
+`<nimble-table-cell>`
+
+-   Renders the data via a custom template
+-   Manages the keyboard interactions to pass focus in/out of custom cells
+
+### Hooking into TanStack Table
+
+We will be using TanStack Table to manage all of the table state related to data, sorting, grouping, hierarchy, filtering, and selection. The TanStack APIs provide callback hooks for when certain aspects of the state are changed (e.g. `onSortingChange`), which allows us to update the UI as needed. A typical lifecycle of a user enacted table state change would be, for example:
+
+-   User clicks header to change sort state
+-   Nimble table callback updates sort state in TanStack Table
+-   TanStack Table issues callback that sort state has changed
+-   Nimble table handles callback and retrieves needed rows from TanStack Table model, and updates the UI
+
+### Hooking into TanStack Virtual
+
+TanStack Virtual provides various pieces of state to enable simple, efficient virtualization of table data. The Nimble Table will provide certain state/objects to the TanStack Virtual API for it to then provide the needed state that we can virtualize the table rows with. Namely:
+
+-   The element that will serve as the scollable element
+-   An estimated height for each row
+-   The total count of rows in the data
+
+With this set of information, the Nimble Table will be able to register a callback to the TanStack Virtual `onChange` which will happen any time the scrollable element scrolls. In that handler the Nimble Table can retrieve the set of virtual items from TanStack Virtual (i.e. `getVirtualItems()`), which represent the total set of rows that should be displayed, and contain the state information that allows the Nimble Table to retrieve the appropriate data from the TanStack Table model to apply to each rendered row, as well as the position each row should be rendered.
+
+_Placeholder for other implementation details_
+
+### Accessibility
+
+The `nimble-table` should align to the grid interaction model provided by the [W3C](https://w3c.github.io/aria-practices/#grid).
+
+### Globalization
+
+The order of table columns is inverted in RTL layouts.
```

Comment:
1. Is this something we're gonig to implement for the first version?
2. How does this affect things like grouping/selection? I.e. will the expander or selection checkbox be on the right as well?

Reply:
> Is there a need to specifically callout theme support in this spec?

I don't believe so, as, ultimately, the Design XD document specifies that implicitly. Other specs haven't explicitly called it out afaik.

Reply:
I've marked the Globalization section as "out of scope" for the initial release. I've also created an open issue regarding your other questions so that we can address them at a later time (this could take the shape of a separate created issue).